packages/formstr-app/src/provider/ApplicationProvider.tsx

```diff
@@ -23,6 +29,13 @@ export const ApplicationProvider: FC<ApplicationProviderProps> = ({
   const [isTemplateModalOpen, setIsTemplateModalOpen] = useState(false);
   const openTemplateModal = () => setIsTemplateModalOpen(true); 
   const closeTemplateModal = () => setIsTemplateModalOpen(false);
+  const navigate = useNavigate();
+  const handleTemplateSelect = (template: FormTemplate) => {
```

Comment:
not sure i like the idea of moving this to applicationProvider, maybe we should have a seperate TemplateProvider?